BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/BeaconUtil.java

```diff
@@ -39,6 +46,126 @@ public static float getAdvertisingRange(int transmissionPower) {
         }
     }
 
+    /**
+     * Gets the smallest distance from the given beacons to the user using a filter.
+     *
+     * @param beaconList Beacons to evaluate the distance from
+     * @param filter     Filter for getting the distance of the beacons
+     * @return Distance to the closest beacon; Double.MAX_VALUE if no beacon was given
+     */
+    public static double getSmallestDistance(List<? extends Beacon> beaconList, WindowFilter filter) {
+        double minimumDistance = Double.MAX_VALUE;
+        for (Beacon beacon : beaconList) {
+            float distance = beacon.getDistance(filter);
+            if (distance < minimumDistance) {
+                minimumDistance = distance;
+            }
+        }
+        return minimumDistance;
+    }
+
+    /**
+     * Gets the closest beacon from the given beacons to the user using a filter.
+     *
+     * @param beaconList Beacons to get the closest one from
+     * @param filter     Filter for getting the distance of the beacons
+     * @return Closest beacon if list is not empty; null else
+     */
+    public static Beacon getClosestBeacon(List<? extends Beacon> beaconList, WindowFilter filter) {
```

Comment:
As the filter is optional, you could also provide a convenience method without a filter that calls `beacon.getDistance()` and thus uses the suggested filter for each beacon.